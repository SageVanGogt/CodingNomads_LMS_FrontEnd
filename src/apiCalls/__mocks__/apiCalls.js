import { mockDocs } from './../../mockData/mockDocs';
import { mockLabs } from './../../mockData/mockLabs';
import { mockTasks } from './../../mockData/mockTasks';
import { mockCourses } from './../../mockData/mockCourses';

export const getAllTasks = jest.fn().mockImplementation(
  () => Promise.resolve(mockTasks)
);

export const getTask = jest.fn();

export const postTask = jest.fn();

export const updateTask = jest.fn();

export const deleteTask = jest.fn();

export const getAllCourses = jest.fn().mockImplementation(
  () => Promise.resolve(mockCourses)
);

export const getCourse = jest.fn();

export const addCourse = jest.fn();

export const updateCourse = jest.fn();

export const deleteCourse = jest.fn();

export const getAllCoursesStudents = jest.fn();

export const addCourseStudent = jest.fn();

export const deleteCourseStudent = jest.fn();

export const getCourseTeachers = jest.fn();

export const addTeacherToCourse = jest.fn();

export const deleteTeacherFromCourse = jest.fn();

export const getCourseTasks = jest.fn();

export const postTaskToCourse = jest.fn();

export const deleteTaskFromCourse = jest.fn();

export const getAllDocs = jest.fn().mockImplementation(
  () => Promise.resolve(mockDocs)
);

export const getAllLabs = jest.fn().mockImplementation(
  () => Promise.resolve(mockLabs)
);

export const deleteLabsFromTask = jest.fn(); 

export const deleteDocsFromTask = jest.fn();

export const deleteTasksFromCourse = jest.fn();

export const deleteStudentsFromCourse = jest.fn();



