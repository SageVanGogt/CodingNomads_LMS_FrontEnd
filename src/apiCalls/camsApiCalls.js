export const getCourseTeachers = async (courseID) => {
  const url = `/api/v1/courses/${courseID}/teachers`;
  try {
    const response = await fetch(url);
    if (response.status !== 200) {
      throw Error('Teachers not found.');
    }
    const teachers = await response.json();
    return teachers;
  } catch (error) {
    throw error;
  }
};

export const addTeacherToCourse = async (courseId, teacherID) => {
  const url = `/api/v1/courses/${courseId}/teachers`;
  const options = {
    method: 'POST',
    headers: { 'content-type': 'application/json' },
    body: JSON.stringify(teacherID)
  };
  try {
    const response = await fetch(url, options);
    if (response.status !== 201) {
      throw Error('Does not have necessary info');
    }
    const course = await response.json();
    return course;
  } catch (error) {
    throw error;
  }
};

export const deleteTeacherFromCourse = async (courseId, teacherId) => {
  const url = `/api/v1/courses/${courseId}/teachers/${teacherId}`;
  const options = {
    method: 'DELETE',
    headers: { 'content-type': 'application/json' }
  };
  try {
    const response = await fetch(url, options);
    if (response.status !== 204) {
      throw Error('That id could not be found.');
    }
  } catch (error) {
    throw error;
  }
};

export const getCourseTasks = async (courseID) => {
  const url = `/api/v1/courses/${courseID}/tasks`;
  try {
    const response = await fetch(url);
    if (response.status !== 200) {
      throw Error('Tasks not found.');
    }
    const teachers = await response.json();
    return teachers;
  } catch (error) {
    throw error;
  }
};

export const addTaskToCourse = async (courseId, taskID) => {
  const url = `/api/v1/courses/${courseId}/tasks`;
  const options = {
    method: 'POST',
    headers: { 'content-type': 'application/json' },
    body: JSON.stringify(taskID)
  };
  try {
    const response = await fetch(url, options);
    if (response.status !== 201) {
      throw Error('Does not have necessary info');
    }
    const course = await response.json();
    return course;
  } catch (error) {
    throw error;
  }
};