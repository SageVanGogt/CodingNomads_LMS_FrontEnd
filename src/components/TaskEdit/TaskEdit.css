.TaskEdit_page {
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: #8E8F8F;
  width: 50vw;
  height: 600px;
  margin-left: 200px; }

.TaskEdit_form {
  display: flex;
  flex-direction: column;
  width: 50%;
  color: white; }

.TaskEdit_input {
  margin-top: 0;
  height: 50px;
  width: 300px;
  border-radius: 10px;
  background-color: #C7C7C7; }

.TaskEdit_submit {
  width: 300px; }

.TaskEdit_select-list {
  width: 300px;
  border: 1px solid black; }
