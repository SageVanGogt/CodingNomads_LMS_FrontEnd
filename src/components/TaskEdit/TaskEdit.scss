.TaskEdit_page {
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: #efefef;
  width: calc(100vw - 200px);
  height: 100vw;
  margin-left: 200px;
  border-radius: 20px;
}

.TaskEdit_title {
  width: 420px;
  text-align: left;
  color: #666666;
  font-size: 1.4rem;
  margin-top: 50px;
}

.TaskEdit_form {
  display: flex;
  flex-direction: column;
  color: white;
}

.TaskEdit_input {
  margin-top: 20px;
  height: 40px;
  width: 400px;
  padding-left: 15px;
  background-color: transparent;
  border: none;
  border-bottom: 3px solid #4885a2;
  font-size: 1.2rem;
  font-family: 'Open Sans', sans-serif;
}

.TaskEdit_input-description {
  background-color: white;
  min-height: 100px;
}

.TaskEdit_submit {
  width: 400px;
}

.TaskEdit_select-list {
  width: 400px;
  margin-top: 30px;
  border: 1px dashed black;
  padding: 20px 0;
}

.TaskEdit_select-list p {
  color: black;
}

.TaskEdit_form label {
  display: none;
}

.TaskEdit_new-select {
  width: 90px;
  height: 20px;
  background-color: #0F94FB;
  box-shadow: 0 5px 6px -4px rgba(0,0,0,0.22), 0 10px 13px -4px rgba(0,0,0,0.34), inset 0 30px 30px 0 rgba(255,255,255,0.35), inset 0 -30px 30px 0 rgba(0,0,0,0.10);
  border-radius: 10px;
}

.TaskEdit_submit {
  height: 30px;
  margin-top: 15px;
  background-color: #0F94FB;  
  box-shadow: 0 5px 6px -4px rgba(0,0,0,0.22), 0 10px 13px -4px rgba(0,0,0,0.34), inset 0 30px 30px 0 rgba(255,255,255,0.35), inset 0 -30px 30px 0 rgba(0,0,0,0.10);
}
