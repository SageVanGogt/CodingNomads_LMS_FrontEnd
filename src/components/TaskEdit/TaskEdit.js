import React, { Component } from 'react';
import { connect } from 'react-redux';
import './TaskEdit.css';
import * as API from '../../apiCalls/apiCalls';
import PropTypes from 'prop-types';
import { LabOptions } from '../LabOptions/LabOptions';
import { DocOptions } from '../DocOptions/DocOptions';

export class TaskEdit extends Component {
  constructor() {
    super();
    this.state = {
      id: null,
      name: '',
      videoLink: '',
      description: '',
      docs: [],
      labs: [],
      allDocs: [],
      allLabs: [],
      docsToDelete: [],
      labsToDelete: [],
      docOptions: [],
      labOptions: []
    };
  }

  componentDidMount = async () => {
    await this.fetchDocs();
    await this.fetchLabs();
  }

  fetchDocs = async () => {
    const docs = await API.getAllDocs();
    this.setState({
      allDocs: docs.data
    });
  }

  fetchLabs = async () => {
    const labs = await API.getAllLabs();
    this.setState({
      allLabs: labs.data
    });
  }

  componentDidUpdate = (prevProps) => {
    const { currentTask } = this.props;
    if (prevProps.currentTask !== currentTask) {
      this.loadTaskInfo(currentTask);
    }
  }

  loadTaskInfo = ({ id, name, videoLink, docs, labs }) => {
    this.setState({
      id,
      name,
      videoLink,
      docs,
      labs
    });
  }

  addDocOptions = (event, docs) => {
    event.preventDefault();

    this.setState({
      docOptions:
        [
          ...this.state.docOptions,
          <DocOptions
            key={`doc-${this.state.docOptions}`}
            id={`doc-option-${this.state.docOptions.length + 1}`}
            docs={docs}
            handleSelectDoc={this.handleSelectDoc}
            deleteDoc={this.deleteDoc}
          />
        ]
    });
  }

  addLabOptions = (event, labs) => {
    event.preventDefault();

    this.setState({
      labOptions:
        [
          ...this.state.labOptions,
          <LabOptions
            key={`lab-${this.state.labOptions}`}
            id={`lab-option-${this.state.labOptions.length + 1}`}
            labs={labs}
            handleSelectLab={this.handleSelectLab}
            deleteLab={this.deleteLab}
          />
        ]
    });
  }

  handleInputChange = (event) => {
    event.preventDefault();
    const { name, value } = event.target;

    this.setState({
      [name]: value
    });
  }

  handleSelectDoc = (event) => {
    event.preventDefault();
    const newDoc = {
      id: event.target.value
    };

    if (!this.state.docs.find(doc => doc.id === newDoc.id)) {
      this.setState({
        docs: [...this.state.docs, newDoc]
      });
    }
  }

  handleSelectLab = (event) => {
    event.preventDefault();
    const newLab = {
      id: event.target.value
    };

    if (!this.state.labs.find(lab => lab.id === newLab.id)) {
      this.setState({
        labs: [...this.state.labs, newLab]
      });
    }
  }

  deleteDoc = (event, key) => {
    event.preventDefault();
    const docId = event.target.previousElementSibling.value;

    if (docId) {
      const updatedDocs =
        this.state.docs.filter(doc => doc.id !== docId);
      const docOptions =
        this.state.docOptions.filter(doc => doc.props.id !== key);
      
      this.setState({
        docs: updatedDocs,
        docOptions,
        docsToDelete: [...this.state.docsToDelete, docId]
      });
    }
  }

  deleteLab = (event, key) => {
    event.preventDefault();
    const labId = event.target.previousElementSibling.value;

    if (labId) {
      const updatedLabs =
        this.state.labs.filter(lab => lab.id !== labId);
      const labOptions =
        this.state.labOptions.filter(lab => lab.props.id !== key);

      this.setState({
        labs: updatedLabs,
        labOptions,
        labsToDelete: [...this.state.labsToDelete, labId]
      });
    }
  }

  determineSubmitMethod = (event) => {
    event.preventDefault();
    if (this.props.currentTask.id) {
      this.patchTask();
    } else {
      this.postNewTask();
    }
  }

  postNewTask = async () => {
    const { name, description, videoLink, docs, labs } = this.state;
    const task = {
      name,
      description,
      videoLink,
      docs,
      labs
    };
    try {
      await API.postTask(task);
    } catch (error) {
      //mdp this error
    }
  }

  patchTask = async () => {
    const { id, name, description, videoLink, docs, labs } = this.state;
    const taskToUpdate = {
      id,
      name,
      description,
      videoLink,
      docs,
      labs
    };

    try {
      await API.updateTask(taskToUpdate);
      this.deleteDocsInDatabase();
      this.deleteLabsInDatabase();
    } catch (error) {
      //mdp this error
    }

  }

  deleteDocsInDatabase = async () => {
    const docsToDelete = this.state.docsToDelete;

    try {
      await API.deleteDocsFromTask(docsToDelete);
    } catch (error) {
      //mdp this error
    }
  }

  deleteLabsInDatabase = async () => {
    const labsToDelete = this.state.labsToDelete;

    try {
      await API.deleteLabsFromTask(labsToDelete);
    } catch (error) {
      //mdp this error
    }
  }

  render() {
    return (
      <div className="TaskCreate_page">
        <form 
          action="submit" 
          className="TaskCreate_form"
          onSubmit={this.determineSubmitMethod}
        >
          <input
            type="text"
            placeholder="name"
            name="name"
            onChange={this.handleInputChange}
            value={this.state.name}
          />
          <input
            type="text"
            placeholder="description"
            name="description"
            onChange={this.handleInputChange}
            value={this.state.description}
          />
          <input
            type="text"
            placeholder="url"
            name="videoLink"
            onChange={this.handleInputChange}
            value={this.state.videoLink}
          />
          <DocOptions
            id={`doc-option-0`}
            docs={this.state.allDocs}
            handleSelectDoc={this.handleSelectDoc}
            deleteDoc={this.deleteDoc}
          />
          {this.state.docOptions}
          <button onClick={(event) => this.addDocOptions(event, this.state.allDocs)}>new doc</button>

          <LabOptions
            id={`lab-option-0`}
            labs={this.state.allLabs}
            handleSelectLab={this.handleSelectLab}
            deleteLab={this.deleteLab}
          />
          {this.state.labOptions}
          <button onClick={(event) => this.addLabOptions(event, this.state.allLabs)}>new lab</button>
          <input type="submit" />
        </form>
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  currentTask: state.currentTask
});

TaskEdit.propTypes = {
  currentTask: PropTypes.object
};

export default connect(mapStateToProps, null)(TaskEdit);
