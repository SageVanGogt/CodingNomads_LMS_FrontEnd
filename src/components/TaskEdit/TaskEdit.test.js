import React from 'react';
import { shallow } from 'enzyme';
import { TaskEdit, mapStateToProps } from './TaskEdit';
import * as API from './../../apiCalls/apiCalls';

jest.mock('./../../apiCalls/apiCalls');

describe('TaskEdit', () => {
  let wrapper;
  let mockCurrentTask;

  beforeEach(() => {
    mockCurrentTask = {id: null, name: ''};
    wrapper = shallow(<TaskEdit />);
  })

  it('should match the snapshot', () => {

  })

  describe('handleChange', () => {
    it('should update the appropriate state', () => {
      let mockEvent = {
        preventDefault: jest.fn(),
        target: {
          name: 'videoLink',
          value: 'caden@youtube'
        }
      };
      let expected = 'caden@youtube';
      wrapper.instance().handleChange(mockEvent);
      let actual = wrapper.state('videoLink');

      expect(actual).toEqual(expected);
    })
  })

  describe('loadTaskInfo', () => {
    it('should update the appropriate state with info from store', () => {
      let mockTask = {
        id: 1,
        name: 'cats',
        videoLink: 'caden@youtube',
        docs: [],
        labs: []
      };
      let expected = 'caden@youtube';
      wrapper.instance().loadTaskInfo(mockTask);
      let actual = wrapper.state('videoLink');

      expect(actual).toEqual(expected);
    })
  })

  describe('handleSubmit', () => {
    it('should call updateTask with the correct params', async () => {
      let mockTask = {
        id: null,
        name: '',
        videoLink: '',
        docs: [],
        labs: []
      };
      let expected = mockTask;
      await wrapper.instance().handleSubmit();

      expect(API.updateTask).toHaveBeenCalledWith(expected);
    })
  })

  describe('mapStateToProps', () =>  {
    it('should return a state with the currentTask prop', () => {
      let mockState = {
        currentTask: {id: null, name: ''},
        user: {id: 1}
      };
      let mappedProps = mapStateToProps(mockState);
      let actual = mappedProps.currentTask;
      let expected = {id: null, name: ''};

      expect(actual).toEqual(expected);
    })
  })
})