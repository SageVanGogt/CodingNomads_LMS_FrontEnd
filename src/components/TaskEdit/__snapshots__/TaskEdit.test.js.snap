// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskEdit should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TaskEdit
    currentTask={
        Object {
            "docs": Array [],
            "id": null,
            "labs": Array [],
            "name": "",
            "videoLink": "",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="TaskEdit_title"
>
          Fill out Task fields
</h1>,
        <form
          action="submit"
          className="TaskEdit_form"
          onSubmit={[Function]}
>
          <label
                    for="name"
          >
                    Task name:
          </label>
          <input
                    className="TaskEdit_input TaskEdit_input-name"
                    name="name"
                    onChange={[Function]}
                    placeholder="name"
                    type="text"
                    value=""
          />
          <label
                    for="description"
          >
                    Description of task:
          </label>
          <textarea
                    className="TaskEdit_input TaskEdit_input-description"
                    name="description"
                    onChange={[Function]}
                    placeholder="description"
                    type="text"
                    value=""
          />
          <label
                    for="videoLink"
          >
                    Link to the lesson video:
          </label>
          <input
                    className="TaskEdit_input TaskEdit_input-videoLink"
                    name="videoLink"
                    onChange={[Function]}
                    placeholder="url"
                    type="text"
                    value=""
          />
          <div
                    className="TaskEdit_select-list"
          >
                    <p>
                              Select documents for your students to read with this task
                    </p>
                    <button
                              className="TaskEdit_new-select"
                              onClick={[Function]}
                    >
                              New Doc +
                    </button>
          </div>
          <div
                    className="TaskEdit_select-list"
          >
                    <p>
                              Select labs for your students to do with this task
                    </p>
                    <button
                              className="TaskEdit_new-select"
                              onClick={[Function]}
                    >
                              New Lab +
                    </button>
          </div>
          <input
                    className="TaskEdit_submit"
                    type="submit"
                    value="Submit Task Changes"
          />
</form>,
      ],
      "className": "TaskEdit_page",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Fill out Task fields",
          "className": "TaskEdit_title",
        },
        "ref": null,
        "rendered": "Fill out Task fields",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "action": "submit",
          "children": Array [
            <label
              for="name"
>
              Task name:
</label>,
            <input
              className="TaskEdit_input TaskEdit_input-name"
              name="name"
              onChange={[Function]}
              placeholder="name"
              type="text"
              value=""
/>,
            <label
              for="description"
>
              Description of task:
</label>,
            <textarea
              className="TaskEdit_input TaskEdit_input-description"
              name="description"
              onChange={[Function]}
              placeholder="description"
              type="text"
              value=""
/>,
            <label
              for="videoLink"
>
              Link to the lesson video:
</label>,
            <input
              className="TaskEdit_input TaskEdit_input-videoLink"
              name="videoLink"
              onChange={[Function]}
              placeholder="url"
              type="text"
              value=""
/>,
            <div
              className="TaskEdit_select-list"
>
              <p>
                            Select documents for your students to read with this task
              </p>
              <button
                            className="TaskEdit_new-select"
                            onClick={[Function]}
              >
                            New Doc +
              </button>
</div>,
            <div
              className="TaskEdit_select-list"
>
              <p>
                            Select labs for your students to do with this task
              </p>
              <button
                            className="TaskEdit_new-select"
                            onClick={[Function]}
              >
                            New Lab +
              </button>
</div>,
            <input
              className="TaskEdit_submit"
              type="submit"
              value="Submit Task Changes"
/>,
          ],
          "className": "TaskEdit_form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Task name:",
              "for": "name",
            },
            "ref": null,
            "rendered": "Task name:",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "TaskEdit_input TaskEdit_input-name",
              "name": "name",
              "onChange": [Function],
              "placeholder": "name",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Description of task:",
              "for": "description",
            },
            "ref": null,
            "rendered": "Description of task:",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "TaskEdit_input TaskEdit_input-description",
              "name": "description",
              "onChange": [Function],
              "placeholder": "description",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "textarea",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Link to the lesson video:",
              "for": "videoLink",
            },
            "ref": null,
            "rendered": "Link to the lesson video:",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "TaskEdit_input TaskEdit_input-videoLink",
              "name": "videoLink",
              "onChange": [Function],
              "placeholder": "url",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  Select documents for your students to read with this task
</p>,
                Array [],
                <button
                  className="TaskEdit_new-select"
                  onClick={[Function]}
>
                  New Doc +
</button>,
              ],
              "className": "TaskEdit_select-list",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Select documents for your students to read with this task",
                },
                "ref": null,
                "rendered": "Select documents for your students to read with this task",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "New Doc +",
                  "className": "TaskEdit_new-select",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "New Doc +",
                "type": "button",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  Select labs for your students to do with this task
</p>,
                Array [],
                <button
                  className="TaskEdit_new-select"
                  onClick={[Function]}
>
                  New Lab +
</button>,
              ],
              "className": "TaskEdit_select-list",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Select labs for your students to do with this task",
                },
                "ref": null,
                "rendered": "Select labs for your students to do with this task",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "New Lab +",
                  "className": "TaskEdit_new-select",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "New Lab +",
                "type": "button",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "TaskEdit_submit",
              "type": "submit",
              "value": "Submit Task Changes",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="TaskEdit_title"
>
            Fill out Task fields
</h1>,
          <form
            action="submit"
            className="TaskEdit_form"
            onSubmit={[Function]}
>
            <label
                        for="name"
            >
                        Task name:
            </label>
            <input
                        className="TaskEdit_input TaskEdit_input-name"
                        name="name"
                        onChange={[Function]}
                        placeholder="name"
                        type="text"
                        value=""
            />
            <label
                        for="description"
            >
                        Description of task:
            </label>
            <textarea
                        className="TaskEdit_input TaskEdit_input-description"
                        name="description"
                        onChange={[Function]}
                        placeholder="description"
                        type="text"
                        value=""
            />
            <label
                        for="videoLink"
            >
                        Link to the lesson video:
            </label>
            <input
                        className="TaskEdit_input TaskEdit_input-videoLink"
                        name="videoLink"
                        onChange={[Function]}
                        placeholder="url"
                        type="text"
                        value=""
            />
            <div
                        className="TaskEdit_select-list"
            >
                        <p>
                                    Select documents for your students to read with this task
                        </p>
                        <button
                                    className="TaskEdit_new-select"
                                    onClick={[Function]}
                        >
                                    New Doc +
                        </button>
            </div>
            <div
                        className="TaskEdit_select-list"
            >
                        <p>
                                    Select labs for your students to do with this task
                        </p>
                        <button
                                    className="TaskEdit_new-select"
                                    onClick={[Function]}
                        >
                                    New Lab +
                        </button>
            </div>
            <input
                        className="TaskEdit_submit"
                        type="submit"
                        value="Submit Task Changes"
            />
</form>,
        ],
        "className": "TaskEdit_page",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Fill out Task fields",
            "className": "TaskEdit_title",
          },
          "ref": null,
          "rendered": "Fill out Task fields",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "action": "submit",
            "children": Array [
              <label
                for="name"
>
                Task name:
</label>,
              <input
                className="TaskEdit_input TaskEdit_input-name"
                name="name"
                onChange={[Function]}
                placeholder="name"
                type="text"
                value=""
/>,
              <label
                for="description"
>
                Description of task:
</label>,
              <textarea
                className="TaskEdit_input TaskEdit_input-description"
                name="description"
                onChange={[Function]}
                placeholder="description"
                type="text"
                value=""
/>,
              <label
                for="videoLink"
>
                Link to the lesson video:
</label>,
              <input
                className="TaskEdit_input TaskEdit_input-videoLink"
                name="videoLink"
                onChange={[Function]}
                placeholder="url"
                type="text"
                value=""
/>,
              <div
                className="TaskEdit_select-list"
>
                <p>
                                Select documents for your students to read with this task
                </p>
                <button
                                className="TaskEdit_new-select"
                                onClick={[Function]}
                >
                                New Doc +
                </button>
</div>,
              <div
                className="TaskEdit_select-list"
>
                <p>
                                Select labs for your students to do with this task
                </p>
                <button
                                className="TaskEdit_new-select"
                                onClick={[Function]}
                >
                                New Lab +
                </button>
</div>,
              <input
                className="TaskEdit_submit"
                type="submit"
                value="Submit Task Changes"
/>,
            ],
            "className": "TaskEdit_form",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Task name:",
                "for": "name",
              },
              "ref": null,
              "rendered": "Task name:",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "TaskEdit_input TaskEdit_input-name",
                "name": "name",
                "onChange": [Function],
                "placeholder": "name",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Description of task:",
                "for": "description",
              },
              "ref": null,
              "rendered": "Description of task:",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "TaskEdit_input TaskEdit_input-description",
                "name": "description",
                "onChange": [Function],
                "placeholder": "description",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "textarea",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Link to the lesson video:",
                "for": "videoLink",
              },
              "ref": null,
              "rendered": "Link to the lesson video:",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "TaskEdit_input TaskEdit_input-videoLink",
                "name": "videoLink",
                "onChange": [Function],
                "placeholder": "url",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    Select documents for your students to read with this task
</p>,
                  Array [],
                  <button
                    className="TaskEdit_new-select"
                    onClick={[Function]}
>
                    New Doc +
</button>,
                ],
                "className": "TaskEdit_select-list",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Select documents for your students to read with this task",
                  },
                  "ref": null,
                  "rendered": "Select documents for your students to read with this task",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "New Doc +",
                    "className": "TaskEdit_new-select",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "New Doc +",
                  "type": "button",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    Select labs for your students to do with this task
</p>,
                  Array [],
                  <button
                    className="TaskEdit_new-select"
                    onClick={[Function]}
>
                    New Lab +
</button>,
                ],
                "className": "TaskEdit_select-list",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Select labs for your students to do with this task",
                  },
                  "ref": null,
                  "rendered": "Select labs for your students to do with this task",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "New Lab +",
                    "className": "TaskEdit_new-select",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "New Lab +",
                  "type": "button",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "TaskEdit_submit",
                "type": "submit",
                "value": "Submit Task Changes",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
